coti/crypto_utils.py

```diff
@@ -114,26 +115,88 @@ def build_input_text(plaintext, user_aes_key, sender, contract, func_sig, signin
     ciphertext, r = encrypt(user_aes_key, plaintext_bytes)
     ct = ciphertext + r
 
-    # Create the function signature
-    func_hash = get_func_sig(func_sig)
     # Sign the message
-    signature = sign_input_text(sender_address_bytes, contract_address_bytes, func_hash, ct, signing_key)
+    signature = sign_input_text(sender_address_bytes, contract_address_bytes, function_selector, ct, signing_key)
 
     # Convert the ct to an integer
     int_cipher_text = int.from_bytes(ct, byteorder='big')
 
-    return int_cipher_text, signature
+    return {
+        'ciphertext': int_cipher_text,
+        'signature': signature
+    }
+
+
+def build_string_input_text(plaintext, user_aes_key, sender, contract, function_selector, signing_key):
+    input_text = {
+        'ciphertext': {
+            'value': []
+        },
+        'signature': []
+    }
 
+    encoded_plaintext = bytearray(list(plaintext.encode('utf-8')))
 
-def build_string_input_text(plaintext, user_aes_key, sender, contract, func_sig, signing_key):
-    encoded_plaintext = array('B', plaintext.encode('utf-8'))
-    encrypted_str = [{'ciphertext': 0, 'signature': b''} for _ in range(len(encoded_plaintext))]
-    for i in range(len(encoded_plaintext)):
-        ct_int, signature = build_input_text(int(encoded_plaintext[i]), user_aes_key, sender, contract,
-                                             func_sig, signing_key)
-        encrypted_str[i] = {'ciphertext': ct_int, 'signature': signature}
+    for i in range(ceil(len(encoded_plaintext) / 8)):
+        start_idx = i * 8
+        end_idx = min(start_idx + 8, len(encoded_plaintext))
 
-    return encrypted_str
+        byte_arr = encoded_plaintext[start_idx:end_idx] + bytearray(8 - (end_idx - start_idx))
+
+        it_int = build_input_text(
+            int.from_bytes(byte_arr, 'big'),
+            user_aes_key,
+            sender,
+            contract,
+            function_selector,
+            signing_key
+        )
+
+        input_text['ciphertext']['value'].append(it_int['ciphertext'])
+        input_text['signature'].append(it_int['signature'])
+    
+    return input_text
+
+def build_address_input_text(plaintext, user_aes_key, sender, contract, function_selector, signing_key):
+    it_int_1 = build_input_text(
```

Comment:
shouldnt we confirm that addres is valid as we did in ts?

Reply:
good idea